spec/fixtures/custom_plugins/kong/plugins/secret-response/handler.lua

```diff
@@ -0,0 +1,21 @@
+local kong_meta = require "kong.meta"
+local decode = require "cjson".decode
+
+
+local SecretResponse = {
+  PRIORITY = 529,
+  VERSION = kong_meta.version,
+}
+
+
+function SecretResponse:access()
+  local reference = kong.request.get_query_arg("reference")
+  local resp, err = kong.vault.get(reference)
+  if not resp then
+    return kong.response.exit(400, { message = err })
+  end
+  return kong.response.exit(200, decode(resp))
+end
```

Comment:
Nice idea with this one! I am wishing I would have thought of this for my other tests.